pandas/core/frame.py

```diff
@@ -3827,15 +3827,8 @@ def _getitem_multilevel(self, key):
         if isinstance(loc, (slice, np.ndarray)):
             new_columns = self.columns[loc]
             result_columns = maybe_droplevels(new_columns, key)
-            if self._is_mixed_type:
-                result = self.reindex(columns=new_columns)
-                result.columns = result_columns
-            else:
-                new_values = self.values[:, loc]
```

Comment:
I guess this is faster than iloc? Did not profile but that's the only reason I can think of that it is done this way. To preserve the performance, I guess we can check something like _can_fast_transpose here as well?

Reply:
But in the case where this matters, I expect that `loc`is a slice? And then I would expect `iloc` to be fast as well? (it has separate paths for single blocks AFAIR) 
But yes, something to verify (indexing the numpy array will _always_ be faster since you don't have the overhead of our indexing machinery. But if we also end up doing the same, the difference should be acceptable).

Reply:
(and I think a bit of extra overhead is also acceptable to avoid needing the custom `add_references` code here (as in https://github.com/pandas-dev/pandas/pull/51944)

Reply:
Sounds good as long as is not to much overhead. We should check at least. That was my reasoning for adding the reference code instead of switching to iloc